test/e2e/prometheus_test.go

```diff
@@ -64,6 +66,31 @@ func TestPrometheusMetrics(t *testing.T) {
 	}
 }
 
+// TestPrometheusGOGC verifies that Prometheus containers use the appropriate gogc value. It also helps us stay informed
+// about any changes to the default gogc value upstream. If that value changes, this test will fail, prompting us to
+// consider whether the new default value is still suitable.
+// Refer to this link for some points that may need to be examined https://github.com/openshift/prometheus/pull/206#issuecomment-2182168575.
+func TestPrometheusGOGC(t *testing.T) {
+	infra, err := f.OperatorClient.GetInfrastructure(ctx, "cluster")
+	require.NoError(t, err)
+	infraConfig := operator.NewInfrastructureConfig(infra)
+
+	gogc := 100
+	if infraConfig.HighlyAvailableInfrastructure() {
+		gogc = 75
+	}
+	f.ThanosQuerierClient.WaitForQueryReturn(
+		t, 5*time.Minute, `min(go_gc_gogc_percent{namespace="openshift-monitoring", service="prometheus-k8s", container="kube-rbac-proxy"})`, // kube-rbac-proxy exposes prometheus container's metrics.
```

Comment:
checked with 4.17.0-0.nightly-2024-06-25-162526, did not find result for `go_gc_gogc_percent{namespace="openshift-monitoring", service="prometheus-k8s", container="kube-rbac-proxy"}`
see
```
$ token=`oc create token prometheus-k8s -n openshift-monitoring`
$ oc -n openshift-monitoring exec -c prometheus prometheus-k8s-0 -- curl -k -H "Authorization: Bearer $token" 'https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query?' --data-urlencode 'query=go_gc_gogc_percent{namespace="openshift-monitoring", service="prometheus-k8s", container="kube-rbac-proxy"}' | jq
{
  "status": "success",
  "data": {
    "resultType": "vector",
    "result": [],
    "analysis": {}
  }
}
```
found below services
```
$ oc -n openshift-monitoring exec -c prometheus prometheus-k8s-0 -- curl -k -H "Authorization: Bearer $token" 'https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query?' --data-urlencode 'query=count (go_gc_gogc_percent{namespace="openshift-monitoring"}) by (service, container, pod)' | jq
{
  "status": "success",
  "data": {
    "resultType": "vector",
    "result": [
      {
        "metric": {
          "container": "cluster-monitoring-operator",
          "pod": "cluster-monitoring-operator-8664984bcd-glnpt",
          "service": "cluster-monitoring-operator"
        },
        "value": [
          1719480396.105,
          "1"
        ]
      },
      {
        "metric": {
          "container": "kube-rbac-proxy-metrics",
          "pod": "thanos-querier-6c756f5fc4-tqvzz",
          "service": "thanos-querier"
        },
        "value": [
          1719480396.105,
          "1"
        ]
      },
      {
        "metric": {
          "container": "kube-rbac-proxy-metrics",
          "pod": "thanos-querier-6c756f5fc4-nckzk",
          "service": "thanos-querier"
        },
        "value": [
          1719480396.105,
          "1"
        ]
      },
      {
        "metric": {
          "container": "kube-rbac-proxy-thanos",
          "pod": "prometheus-k8s-1",
          "service": "prometheus-k8s-thanos-sidecar"
        },
        "value": [
          1719480396.105,
          "1"
        ]
      },
      {
        "metric": {
          "container": "kube-rbac-proxy-thanos",
          "pod": "prometheus-k8s-0",
          "service": "prometheus-k8s-thanos-sidecar"
        },
        "value": [
          1719480396.105,
          "1"
        ]
      },
      {
        "metric": {
          "container": "metrics-server",
          "pod": "metrics-server-6fbd55df99-qlb8q",
          "service": "metrics-server"
        },
        "value": [
          1719480396.105,
          "1"
        ]
      },
      {
        "metric": {
          "container": "metrics-server",
          "pod": "metrics-server-6fbd55df99-tdt2m",
          "service": "metrics-server"
        },
        "value": [
          1719480396.105,
          "1"
        ]
      }
    ],
    "analysis": {}
  }
}
```

Reply:
Forgot to [mention it](https://github.com/openshift/prometheus/pull/206#issuecomment-2189165997) here, the metric was added in prometheus 2.53.0, but the newest nightly builds are still using prometheus 2.52.0 (you can check that `prometheus --version` inside the prom container), the version bump https://github.com/openshift/prometheus/pull/206 wasn't included yet.
(apparently the latest builds are failing)

Reply:
You can test it on the latest 4.17.0-0.ci as it'll be done by ci/prow/e2e-agnostic-operator